src/drivers/intel/ssp/ssp.c

```diff
@@ -741,20 +741,45 @@ static int ssp_get_hw_params(struct dai *dai,
 	return 0;
 }
 
-/* SSP hw params */
+/*
+ * enabled clock MCLK/BCLK to support codec that need MCLK/BCLK
+ * This is called before triiger(COMP_TRIGGER_START)
+ */
 static int ssp_hw_params(struct dai *dai,
 			 struct sof_ipc_stream_params  *params)
 {
 	dai_info(dai, "ssp_hw_params()");
 
```

Comment:
do we need to check clk_early_start flag here and in ssp_hw_free since they are always called regardless the flag?

Reply:
> do we need to check clk_early_start flag here and in ssp_hw_free since they are always called regardless the flag?

Yes, updated, thanks.